courses/views_test.py

```diff
@@ -239,3 +249,87 @@ def test_user_exists(self):
         resp = self.client.get(reverse('user-existence'), {'uid': author.edx_uid})
         self.assertEqual(resp.status_code, 200)
         self.assertEqual(json.loads(resp.content.decode('utf-8')), {'exists': True})
+
+
+# pylint: disable=no-self-use
+class CCXCreateTests(ApiTests):
+    """Test CCX create API for making CCXs on edx"""
+
+    def setUp(self):
+        self.payload = {
+            'master_course_id': 'foo',
+            'user_email': 'bleh@example.com',
+            'total_seats': 102,
+            'display_name': 'test name',
+        }
+        super(CCXCreateTests, self).setUp()
+
+    def test_missing_arguments_throw_error(self):
+        """Throw an error if we're missing a POST parameter"""
+        for key in ['master_course_id', 'user_email', 'total_seats', 'display_name']:
+            payload = self.payload.copy()
+            del payload[key]
+
+            result = self.client.post(reverse('create-ccx'), payload)
+
+            assert result.status_code == 400, str(result.content)
+            assert re.search('POST argument {}'.format(key).encode(), result.content)
+
+    def test_unknown_course_throws_404(self):
+        """If we're given an unknown master_course_id, throw a 404"""
+        result = self.client.post(reverse('create-ccx'), self.payload)
+
+        assert result.status_code == 404, str(result.content)
+
+    def test_request_error_returns_error(self):
+        """If there's an error with the edx request, return an error"""
+        course = CourseFactory.create()
+        self.payload['master_course_id'] = course.course_id
+
+        with mock.patch('courses.views.requests', autospec=True) as mock_req:
+            mock_req.post.side_effect = RequestException
+
+            result = self.client.post(reverse('create-ccx'), self.payload)
+
+            assert result.status_code == 502, str(result.content)
```

Comment:
@bdero is there any significance to returning 5xx error codes? I know he gets emails for exceptions resulting in 500 errors, not sure if that extends to cases where we return it purposefully.